tensorboard/functionaltests/wct_test_driver.py

```diff
@@ -90,13 +121,26 @@ def tearDown(cls):
       cls.process.wait()
 
     def test(self):
-      driver = webtest.new_webdriver_session()
+      driver = webtest.new_webdriver_session(
+          capabilities={"loggingPrefs": {"browser": "ALL"}}
+      )
       url = "http://localhost:%s%s" % (self.port, web_path)
       driver.get(url)
-      wait.WebDriverWait(driver, 10).until(
-          expected_conditions.title_contains("test"))
-      title = driver.title
-      if "failing test" in title or "passing test" not in title:
-        self.fail(title)
+      browser_log_indicates_result = BrowserLogIndicatesResult()
+      try:
+        wait.WebDriverWait(driver, 10).until(browser_log_indicates_result)
+        if not browser_log_indicates_result.passed:
+          self.fail()
+      finally:
+        # Print log as an aid for debugging.
+        log = browser_log_indicates_result.log + driver.get_log("browser")
+        self._printLog(log)
```

Comment:
Nice; thanks! It looks like this prints the assertion failures
(“expected 0, got 1”), which is definitely better than the previous
behavior even though the test case name isn’t logged.

Reply:
Yep. It would be nice to try to get that as well, but I'll stop here for now.
PTAL. Thanks!

Reply:
Note that I don't have write-access and cannot merge.